src/playback-watcher.js

```diff
@@ -107,18 +131,81 @@ export default class PlaybackWatcher {
 
     const currentTime = this.tech_.currentTime();
     const buffered = this.tech_.buffered();
+    const atEndBuffer = !buffered.length ||
+      currentTime + Ranges.SAFE_TIME_DELTA >= buffered.end(buffered.length - 1);
+    // If current time is at the end of the final buffered region, then any playback
+    // stall is most likely caused by buffering in a low bandwidth environment. The tech
+    // should fire a `waiting` event in this scenario, but due to browser and tech
+    // inconsistencies. Calling `techWaiting_` here allows us to simulate
+    // responding to a native `waiting` event when the tech fails to emit one.
+
+    if (this.lastRecordedTime === currentTime && atEndBuffer) {
+      if (this.techWaiting_()) {
+        return;
+      }
+      const mpc = this.vhs.masterPlaylistController_;
+      const previousBytes = this.stallStartBytes_;
+
+      if (!previousBytes) {
+        this.tech_.off('timeupdate', this.checkForStallFix_);
+        this.tech_.on('timeupdate', this.checkForStallFix_);
+      }
+
+      this.stallStartBytes_ = mpc.mediaBytesTransferred_();
```

Comment:
Although it's a less likely scenario, we may also want to consider the case where the VTT loader is infinitely downloading as well, as that was an issue we recently resolved, but is possible may recur.

Reply:
Do you have a link to the pull request and the content that I can use for testing this issue? I agree that we should detect it but it may require something a bit different.

Reply:
PR: https://github.com/videojs/http-streaming/pull/825 , stream was the Axinom live HLS stream